src/mbgl/renderer/sources/render_raster_dem_source.cpp

```diff
@@ -55,6 +55,51 @@ void RenderRasterDEMSource::update(Immutable<style::Source::Impl> baseImpl_,
                        [&] (const OverscaledTileID& tileID) {
                            return std::make_unique<RasterDEMTile>(tileID, parameters, *tileset);
                        });
+    auto rendertiles = getRenderTiles();
```

Comment:
`getRenderTiles()` returns only the `RenderTile`s that are about to be rendered. This excludes the previously loaded tiles that may have been cached in `TilePyramid.cache`. 

Iterating all the tiles on each update seems unnecessary. 

Alternatively:
- Overload `onTileLoaded`/`onTileChanged` in `RenderRasterDEMSource` (may require changes in `RenderSource`).
- When a tile is loaded query the `TilePyramid` for its specific neighbors and backfill both. Ensue that the query includes both the `tiles` map and the `TileCache` 
- Change `TileCache` to access the cached tile directly using something like `Tile* get(OverscaledTileID& key) const`. Change the existing `get` method to `pop`

Reply:
👍 done in 0f0aaf1 and 85818c8